src/pyhf/events.py

```diff
@@ -24,19 +23,66 @@ def noop(*args, **kwargs):
     pass
 
 
-class WeakList(list):
-    def append(self, item):
-        list.append(self, weakref.WeakMethod(item, self.remove))
+class Callables:
+    def __init__(self):
+        self._callbacks = []
+
+    @property
+    def callbacks(self):
+        """
+        Get the current list of living callbacks.
+        """
+        self._flush()
+        return self._callbacks
+
+    def append(self, callback):
+        """
+        Append a new bound method as a callback to the list of callables.
+        """
+        try:
+            # methods
+            callback_ref = weakref.ref(callback.__func__), weakref.ref(
+                callback.__self__
+            )
+        except AttributeError:
+            callback_ref = weakref.ref(callback), None
```

Comment:
I'm sure that I'm missing the obvious here, but if this isn't already being hit anywhere in CI then what are the types of events that _would_ trigger this?